### PR TITLE
RHOAIENG-2787: Example of GitOps approach with ArgoCD directly, without ACM

### PR DESCRIPTION
**Pull request by grdryn**
JIRA: https://issues.redhat.com/browse/RHOAIENG-2787

This PR stacks upon #229, and the commits here may change slightly, if there are changes required for that.

## Description

We love OCM/ACM and recommend using it where possible, but if you've got your own multi-cluster and application management solution, and don't want to have to replace it to use the solutions in this repo, then you can instead use this as a starting point to integrate it yourself.

The steps are intended to be documented in such a way that they could be either executed manually or adapted to fit into any system.

## How Has This Been Tested?

- [x] Existing approach using ACM continues to work as expected
- [x] Following the steps in the new docs/byo-mgmt-gitops.md document work as expected:
  - [x] The example applications deploy as expected

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,19 +55,26 @@ contains example models in the [pipelines/models/](pipelines/models/) directory.
 The diagram above illustrates the architecture of the proof of concept. The two main categories of clusters are the core
 and near edge clusters.
 
+> [!NOTE]
+> The rest of this documentation assumes usage of ACM[^2] for cluster and application management, and we highly recommend
+> this approach.
+> If you choose to use a different approach, you'll find some information on how to deploy the inference container
+> application using standalone ArgoCD instances on the edge clusters, in the dedicated
+> [Bring Your Own Cluster Management document](docs/byo-mgmt-gitops.md).
```
Comment (grdryn) [inline README.md:63]: I've added this here and in docs/preparing_the_infrastructure.md to draw attention to the possibility while trying not to pollute the docs with what I (at least currently) consider a second class alternative approach.

File: docs/byo-mgmt-gitops.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+## Observability
+
+On each of the edge clusters, you can enable [monitoring for user-defined projects]:
+* `oc -n openshift-monitoring edit configmap cluster-monitoring-config`
+* Set variable `enableUserWorkload` to `true`.
+
+If you forward metrics from each edge cluster to a central location, you can find a list of example metrics from the
+OpenVINO and Seldon model servers from our examples in
+[metrics_list](acm/odh-core/acm-observability/files/metrics_list.yaml) that you may wish to forward.
```
Comment (grdryn) [inline docs/byo-mgmt-gitops.md:63]: @StevenTobin does this make sense?

File: byo-mgmt/registration/near-edge/base/argocd-application.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+kind: Application
+metadata:
+  name: application
+  namespace: openshift-gitops
+spec:
+  project: default
+  source:
+    path: acm/odh-edge/apps/app
+    repoURL: 'https://github.com/opendatahub-io/ai-edge.git'
+    targetRevision: main
```
Comment (grdryn) [inline byo-mgmt/registration/near-edge/base/argocd-application.yaml:11]: If trying this out locally, one might have more success changing these values to be the `grdryn` fork, and the `RHOAIENG-2787_byo-mgmt` branch from this PR.

File: docs/byo-mgmt-gitops.md (hunk 2 above)
Comment (StevenTobin) [reply docs/byo-mgmt-gitops.md:63]: Yep, this lgtm

**Review by LaVLaS — APPROVED** [review]
/lgtm

Without the namespace label present, I was presented with the error message in the format of 
```
one or more objects failed to apply, reason:
<<OBJECT>> is forbidden: User "system:serviceaccount:openshift-gitops:openshift-gitops-argocd-application-controller" cannot create resource "<<OBJECT>>" in API group "apps" in the namespace "<<APPLICATION TARGET NAMESPACE",...
```

**Pull request merged**